indra/newview/llimprocessing.cpp

```diff
@@ -579,8 +604,16 @@ void LLIMProcessing::processNewMessage(LLUUID from_id,
                             region_message = true;
                         }
                     }
-                    gIMMgr->addMessage(
-                        session_id,
+
+                    if (is_system_notice)
+                    {   // The simulator has injected some sort of notice into the conversation.
+                        // findString will only replace the contents of buffer if the notice_id is found.
+                        LLTrans::findString(buffer, notice_id, notice_args);
+                        name   = SYSTEM_FROM;
+                        from_id = LLUUID::null;
+                    }
```

Comment:
Rather than bother with `is_system_notice` you could pivot on whether `notice_id` is interesting:  `if (notice_id.notNull())`.

Reply:
Your comment also flagged for me that the name `notice_id` may be mistaken to be a UUID. 
Changed to `notice_name` and test for empty.